pkg/advisory/create.go

```diff
@@ -88,9 +88,23 @@ func Create(ctx context.Context, req Request, opts CreateOptions) error {
 }
 
 func createAdvisoryConfig(ctx context.Context, documents *configs.Index[v2.Document], req Request) error {
-	newAdvisoryID, err := GenerateCGAID()
-	if err != nil {
-		return fmt.Errorf("generating CGA ID: %w", err)
+	var newAdvisoryID string
+
+	// Continuously generate new CGA IDs until we get a unique one.
+	for {
+		id, err := GenerateCGAID()
+		if err != nil {
+			return fmt.Errorf("generating CGA ID: %w", err)
+		}
+
+		exists, err := CGAIDExists(ctx, id)
+		if err != nil {
+			return fmt.Errorf("checking existence of %s: %w", id, err)
```

Comment:
```suggestion
			return fmt.Errorf("checking existence of ID %q: %w", id, err)
```

(Just to be clear that we're checking on the ID, not the advisory named by that ID)